File: backup_antes_historico/frontend_lib/models/comment.dart
import 'user.dart';

class TicketCommentModel {
  final int id;
  final int ticketId;
  final String content;
  final DateTime createdAt;
  final User? author;

  const TicketCommentModel({
    required this.id,
    required this.ticketId,
    required this.content,
    required this.createdAt,
    this.author,
  });

  factory TicketCommentModel.fromJson(Map<String, dynamic> json) => TicketCommentModel(
        id: json['id'],
        ticketId: json['ticket_id'],
        content: json['content'],
        createdAt: DateTime.parse(json['created_at']),
        author: json['author'] != null ? User.fromJson(json['author']) : null,
      );
}

File: backend/app/models/comment.py
from datetime import datetime, timezone

from sqlalchemy import DateTime, ForeignKey, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.session import Base


class TicketComment(Base):
    __tablename__ = "ticket_comments"

    id: Mapped[int] = mapped_column(primary_key=True)
    ticket_id: Mapped[int] = mapped_column(ForeignKey("tickets.id"), nullable=False, index=True)
    author_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc), nullable=False
    )

    ticket = relationship("Ticket", back_populates="comments")
    author = relationship("User", back_populates="comments")
